File: 1.two-sum.py
# @leet start
from typing import List

def solve(nums: List[int], target: int) -> List[int]:
    for i in range(len(nums)):
        for j in range(i+1, len(nums)):
            if nums[i] + nums[j] == target:
                return [i, j]
    raise Exception("No solution")
    
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        return solve(nums, target)

# @leet end

File: 1.two-sum.cpp
#include <vector>
using namespace std;
class Solution {
public:
    std::vector<int> twoSum(std::vector<int>& nums, int target) {
        vector<int> res;
        int slow = 0;
        int fast = 1;

        while (slow < fast && slow < nums.size()-1){
            if (nums[fast] + nums[slow] == target)
            {
                res.push_back(slow);
                res.push_back(fast);
                return res;
            }
            if (fast >= nums.size()-1){
                slow++;
                fast = slow+1;
            } else if (fast < nums.size()-1) {
                fast++;
            }
        }
        return res;
    }
};
